include/type.hpp

```diff
@@ -74,11 +80,39 @@ class PtrType : public Type {
   }
 
   bool IsEqual(const Type& that) const noexcept override;
+  std::size_t ToSize() const override;
   std::string ToString() const override;
   std::unique_ptr<Type> Clone() const override;
 
  private:
   std::unique_ptr<Type> base_type_;
+  const std::size_t size_ = 8;
+};
+
+class ArrType : public Type {
+ public:
+  explicit ArrType(std::unique_ptr<Type> base_type, std::size_t len)
+      : base_type_{std::move(base_type)}, len_{len} {}
+
+  const Type& base_type()  // NOLINT(readability-identifier-naming)
+      const noexcept {
+    return *base_type_;
+  }
+
+  bool IsArr() const noexcept override {
+    return true;
+  }
+
+  bool IsEqual(const Type& that) const noexcept override;
+  std::size_t ToSize() const override;
```

Comment:
Since the size of an array is the size of its base type times the number of elements, which may not be intuitive, I think it would be nice if we add a function comment for it.
```suggestion
  /// @note The size of an array is (size of base type) * (number of elements).
  std::size_t ToSize() const override;
```